nerfstudio/viewer/render_state_machine.py

```diff
@@ -168,9 +169,11 @@ def _render_img(self, camera_state: CameraState):
                         with torch.no_grad(), viewer_utils.SetTrace(self.check_interrupt):
                             outputs = self.viewer.get_model().get_outputs_for_camera(camera, obb_box=obb)
                 except viewer_utils.IOChangeException:
-                    self.viewer.get_model().train()
+                    if was_training:
+                        self.viewer.get_model().train()
                     raise
-                self.viewer.get_model().train()
+                if was_training:
```

Comment:
we could do:
```python
finally:
    if was_training:
        self.viewer.get_model().train()
```
and reduce some duplication?

Reply:
I don't think this is possible because of the re-raising behavior of the except? is it possible for the finally to run after raising another exception?

Reply:
Oh interesting it actually is possible in python, just updated it

Reply:
oh interesting good point, TIL also!!